public/app/plugins/panel/pmm-check/components/Failed/Failed.test.tsx

```diff
@@ -118,7 +173,59 @@ describe('Failed::', () => {
     await waitFor(() => expect(screen.getByTestId('db-check-panel-zero-checks')).toBeInTheDocument());
   });
 
-  it('should render a message when the user only has reader access', async () => {
+  it('should render 0 when the sum of all checks is zero for editor user', async () => {
+    config.bootData.user.isGrafanaAdmin = false;
+    config.bootData.user.orgRole = OrgRole.Editor;
+    jest.spyOn(CheckService, 'getAllFailedChecks').mockImplementationOnce(async () => [
+      {
+        serviceName: '',
+        serviceId: '',
+        counts: {
+          emergency: 0,
+          critical: 0,
+          alert: 0,
+          error: 0,
+          warning: 0,
+          notice: 0,
+          info: 0,
+          debug: 0,
+        },
+      },
+      {
+        serviceName: '',
+        serviceId: '',
+        counts: {
+          emergency: 0,
+          critical: 0,
+          alert: 0,
+          error: 0,
+          warning: 0,
+          notice: 0,
+          info: 0,
+          debug: 0,
+        },
+      },
+    ]);
+
+    render(
+      <Provider
+        store={configureStore({
+          percona: {
+            user: { isAuthorized: false },
+            settings: { loading: false, result: { advisorEnabled: true, isConnectedToPortal: false } },
+          },
+        } as StoreState)}
+      >
+        <Failed />
+      </Provider>
+    );
+
+    await waitFor(() => expect(screen.getByTestId('db-check-panel-zero-checks')).toBeInTheDocument());
+  });
+
+  it('should render a message and no failed checks for viewer user', async () => {
```

Comment:
I think the title could be better, something like "should render unauthorised message for viewer user"